chore(deps): Update python dependencies

gavindsouza opened the pull request:
### Updates
* Cryptography (3.4.8 => 37.0.2)
* IPython (7.31.0 => 8.4.0)
* ~Redis (3.5.3 => 4.3.3)~ [Dropped because of blocker - https://github.com/frappe/erpnext/issues/31300)
* Faker (8.1.4 => 13.12.1)
* pdfkit (0.6.1 => 1.0.0)
* PyPDF2 (1.26.0 => 2.1.0)
* pytz (2021.1 => 2022.1)
* croniter (1.0.15 => 1.3.5)
* Jinja2 (3.0.3 => 3.1.2)
* Pillow (9.0.1 => 9.1.1)
* psutil (5.8.0 => 5.9.1)
* requests (2.25.1 => 2.27.1)
* rq (1.8.1 => 1.10.1)
* semantic-version (2.8.5 => 2.10.0)
* Werkzeug (2.0.3 => 2.1.2)


### Changes

* Clearer error message when missing / invalid encryption_key in conf
* API deprecations in PyPDF2
* API updates in pdfkit
* Deprecation of dependency_links (setuptools)
* Constants deprecation in Pillow

gavindsouza committed:
chore(deps): Bump requests from 2.25.1 to 2.27.1

gavindsouza committed:
chore(deps): Bump Pillow from 9.0.1 to 9.1.1

Includes security fixes, API & constants deprecations

Changelog: https://pillow.readthedocs.io/en/stable/releasenotes/index.html

gavindsouza committed:
chore(deps): Bump semantic-version from 2.8.5 to 2.10.0

gavindsouza committed:
chore(deps): Bump croniter from 1.0.15 to 1.3.5

changelog: https://pypi.org/project/croniter/#changelog

gavindsouza committed:
chore(deps): Bump cryptography from 3.4.8 to 37.0.2

changelog: https://cryptography.io/en/latest/changelog/
diff: https://github.com/pyca/cryptography/compare/3.4.8...37.0.x

gavindsouza committed:
fix: Cleaner error message on invalid encryption_key

gavindsouza committed:
chore(deps): Bump IPython from 7.31.0 to 8.4.0

Major version includes multiple bug, security & performance fixes

changelog: https://ipython.readthedocs.io/en/stable/whatsnew/version8.html

gavindsouza commented:
The newer IPython feels a lot more snappy 😄

ankush commented:
Few more safe updates, that I had verified a while back. 

PS: bleach has kinda major breaking change in 5.x so sticking to 4.x


```diff
diff --git a/requirements.txt b/requirements.txt
index bf798fe747..987faf9f50 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,12 +1,12 @@
-Babel~=2.9.0
-beautifulsoup4~=4.9.3
+Babel~=2.10.1
+beautifulsoup4~=4.11.1
 bleach-allowlist~=1.0.3
-bleach~=3.3.0
+bleach~=4.1.0
 boto3~=1.17.53
 braintree~=4.8.0
 chardet~=4.0.0
 Click~=7.1.2
-croniter~=1.0.11
+croniter~=1.3.5
 cryptography~=3.4.7
 dropbox~=11.7.0
 email-reply-parser~=0.5.12
```


https://babel.pocoo.org/en/latest/changelog.html
https://bazaar.launchpad.net/~leonardr/beautifulsoup/bs4/view/head:/CHANGELOG
https://bleach.readthedocs.io/en/latest/changes.html
https://pypi.org/project/croniter/#changelog

Alchez commented:
@gavindsouza, can we update Click as well to >8.0? It's not required by Frappe directly, but [Black's requirements](https://github.com/psf/black/blob/main/.github/mypyc-requirements.txt) cause it to sometimes stop working.

ankush commented:
click 8 and 8.1 have big list of possibly breaking changes 🥲 but yeah, we should upgrade it and upgrade our black fork too. 


ref: https://click.palletsprojects.com/en/8.1.x/changes/

gavindsouza commented:
These major version updates are pending to be verified:

```python
 % python3 pipupgrade.py --major
bleach: 3.3.1 => 5.0.0
google-auth: 1.29.0 => 2.7.0
pyOpenSSL: 20.0.1 => 22.0.0
redis: 3.5.3 => 4.3.3
stripe: 2.56.0 => 3.3.0

coverage: 5.5 => 6.4.1
PyYAML: 5.4.1 => 6.0
WeasyPrint: 52.5 => 55.0
```

There's minor updates available too, but they're hopefully going to be relatively no pain in moving up:

```python
 % python3 pipupgrade.py --minor
Babel: 2.9.1 => 2.10.1
beautifulsoup4: 4.9.3 => 4.11.1
boto3: 1.17.112 => 1.24.5
braintree: 4.8.0 => 4.15.2
cairocffi: 1.2.0 => 1.3.0
dropbox: 11.7.0 => 11.31.0
google-api-python-client: 2.2.0 => 2.50.0
google-auth-oauthlib: 0.4.6 => 0.5.1
googlemaps: 4.4.7 => 4.6.0
oauthlib: 3.1.1 => 3.2.0
premailer: 3.8.0 => 3.10.0
pycryptodome: 3.10.4 => 3.14.1
PyJWT: 2.0.1 => 2.4.0
pyngrok: 5.0.6 => 5.1.0
razorpay: 1.2.0 => 1.3.0
unittest-xml-reporting: 3.0.4 => 3.2.0
```

---

Will likely skip the libraries of integrations altogether - ohwell, let's see how it goes.

gavindsouza committed:
chore(deps): bump faker from 8.1.4 to 13.12.1

Faker is a dev dependency installed for tests - to access the
frappe.mock API to generate paragraphs, names & emails.

gavindsouza committed:
chore: Remove deprecated dependency_links from setup

* pdfkit dep is already being fulfilled by requirements.txt
* ref: https://stackoverflow.com/questions/12518499/pip-ignores-dependency-links-in-setup-py

gavindsouza committed:
chore(deps): Bumped pdfkit from 0.6.1 to 1.0.0

* Updated from_string API to match latest signature
* changelog: https://github.com/JazzCore/python-pdfkit/blob/master/HISTORY.rst

gavindsouza committed:
chore(deps): Bump PyPDF2 from 1.26.0 to 2.1.0

* Updated changes in API usages
* changelog: https://github.com/py-pdf/PyPDF2/blob/main/CHANGELOG

gavindsouza committed:
chore(deps): Bump pytz from 2021.1 to 2022.1

Multiple bug fixes & IANA 2022a

* changelog: https://github.com/stub42/pytz/blob/master/tz/NEWS
* diff: https://github.com/stub42/pytz/compare/release_2021.1...release_2022.1

gavindsouza committed:
chore(deps): Bump rq from 1.8.1 to 1.10.1

Exciting improvements to rq! Cleanup & bugs that made us sccratch heads
in prod.

changelog: https://github.com/rq/rq/blob/master/CHANGES.md

gavindsouza committed:
chore(deps): Bump Jinja2 from 3.0.3 to 3.1.2

changelog: https://jinja.palletsprojects.com/en/3.1.x/changes/

gavindsouza committed:
chore(deps): Bump psutil from 5.8.0 to 5.9.1

gavindsouza committed:
chore(deps): Bump Werkzeug from 2.0.3 to 2.1.2

changelog: https://werkzeug.palletsprojects.com/en/2.1.x/changes/

gavindsouza commented:
Updating 14 dependencies via this PR. My previous comment shows the list of available updates. They contain deprecations and API changes too. I can't verify and update them now so I'll leave them for another time - or for someone else :trollface:

gavindsouza committed:
Merge branch 'develop' into update-python-deps
